File: examPortal/src/main/java/com/exam/Service/QuizService.java
package com.exam.Service;

import java.util.List;
import java.util.Set;

import com.exam.entity.Category;
import com.exam.entity.Quiz;

public interface QuizService {
	
	public Quiz addQuiz(Quiz quiz);
	
	public Quiz updateQuiz(Quiz quiz);
	
	public List<Quiz> getAllQuiz();
	
	public Quiz getQuiz(Long quizId);
	
	public void deleteQuiz(Long quizId);
	
	public List<Quiz> getQuizOfCategory(Category cat);
	
	public List<Quiz> getAllActiveQuiz();

}

File: examFront/src/app/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private http:HttpClient) { }

  public getAllQuiz(){
    return this.http.get(`${baseUrl}/quiz/`);
  }

  public saveQuiz(quizData:any){
   return this.http.post(`${baseUrl}/quiz/save`,quizData);
  }

  public deleteQuiz(qId:number){
   return this.http.delete(`${baseUrl}/quiz/${qId}`);
  }

  public getSingleQuiz(qId:number){
  return this.http.get(`${baseUrl}/quiz/${qId}`);
  }

  public updateQuiz(quizData:any){
    return this.http.put(`${baseUrl}/quiz/`,quizData);
  }

  public getQuizByCategory(qId:number){
      return this.http.get(`${baseUrl}/quiz/category/${qId}`);
  }

  public getAllActiveQuiz(){
      return  this.http.get(`${baseUrl}/quiz/activeQuiz`);
  }
}
